isabelle/haskell/MarloweCoreJson.lhs

```diff
@@ -0,0 +1,1394 @@
+                                                        <*> getInteger "but only paid" butOnlyPaid
+                                                        <*> (v .: "asked_to_pay"))
+    <|> (TransactionShadowing <$> (v .: "value_id")
+                              <*> (v .: "had_value")
+                              <*> (v .: "is_now_assigned"))
+  parseJSON _ =
+    fail "Contract must be either an object or a the string \"close\""
+\end{code}
+
+Some examples for each \emph{TransactionWarning} type
+
+
+\isamarkupsubsubsection{TransactionNonPositiveDeposit}
+
+\begin{code}
+transactionNonPositiveDepositExample :: TransactionWarning
+transactionNonPositiveDepositExample = TransactionNonPositiveDeposit
+                                            addressExample
+                                            roleExample
+                                            dolarToken
+                                            20
+\end{code}
+
+is serialized as
+\perform{prettyEncodeExample transactionNonPositiveDepositExample}
+
+
+\isamarkupsubsubsection{TransactionNonPositivePay}
+
+\begin{code}
+transactionNonPositivePayExample :: TransactionWarning
+transactionNonPositivePayExample = TransactionNonPositivePay
+                                            addressExample
+                                            internalPayeeExample
+                                            dolarToken
+                                            20
+\end{code}
+
+is serialized as
+\perform{prettyEncodeExample transactionNonPositivePayExample}
+
+
+\isamarkupsubsubsection{TransactionPartialPay}
+
+\begin{code}
+transactionPartialPayExample :: TransactionWarning
+transactionPartialPayExample = TransactionPartialPay
+                                            addressExample
+                                            internalPayeeExample
+                                            dolarToken
+                                            20
+                                            30
+\end{code}
+
+is serialized as
+\perform{prettyEncodeExample transactionPartialPayExample}
+
+
+\isamarkupsubsubsection{TransactionShadowing}
+
+\begin{code}
+transactionShadowingExample :: TransactionWarning
+transactionShadowingExample = TransactionShadowing
+                                            (ValueId "example")
+                                            4
+                                            5
+\end{code}
+
+is serialized as
+\perform{prettyEncodeExample transactionShadowingExample}
+
+
+\isamarkupsubsubsection{TransactionAssertionFailed}
+
+\begin{code}
+transactionAssertionFailedExample :: TransactionWarning
+transactionAssertionFailedExample = TransactionAssertionFailed
+\end{code}
+
+is serialized as \eval{encodeExample transactionAssertionFailedExample}
+
+
+\isamarkupsection{IntervalError}
+
+The \emph{IntervalError} type is serialized as an object with a single property (depending on the constructor)
+and in a tuple, the values.
```

Comment:
```suggestion
The \emph{IntervalError} type is serialized as an object with a single property (depending on the constructor)
and an array with the values.
```

Reply:
Hm, we should have properties for each of the values, because arrays are for when we don't know the number of elements in advance and they all mean the same. In this case, each value has a different meaning, so having a label would make it clearer to the reader and also probably translates in a better JSON Schema

Reply:
This is how its currently serialized in `purescript-marlowe` and `marlowe-cardano`. I don't like it either, that is why I mentioned it in the PR description. I can rename them and then we need to create a ticket to change it in both places.